include/circt/Dialect/Pipeline/Pipeline.td

```diff
@@ -161,6 +157,22 @@ def ScheduledPipelineOp : Op<Pipeline_Dialect, "scheduled", [
     // This enables certain invariants such as "all values used within a stage
     // must be defined within a stage".
     bool isMaterialized();
+
+    // Returns the arguments for a stage.
+    ValueRange getStageArguments(Block* stage) {
+      if(stage != getEntryStage())
+        return stage->getArguments();
+
+      // The entry block defines a lot of SSA values...
+      // drop everything but the 'input's to the stage
+      return stage->getArguments().take_front(getInputs().size());
```

Comment:
I am uncomfortable with making the assumption that the inputs are the first group of block args, but I don't think there's any alternative. Matt was commenting on something similar (having to rely on port numbers rather than named symbols). You might think on a solution for both and if the problems relate.

Just something I want to put in your brain.

Reply:
For this, is there anything that ODS generates which could be used to get the offset? (Which would be zero with the current ODS def.)

Reply:
Changed to
>       return stage->getArguments().slice(getInputs().getBeginOperandIndex(), getInputs().size());